File: data/code_pairs/cpp/1372.cpp
#ifndef RD3DMODL_GENMOD_H
#define RD3DMODL_GENMOD_H

#include <string>
#include <vector>

namespace RD3DMODL__genmod {

void RD3DMODL(const std::string& FILENAME, int& IERROR, const int NMODKERN, const int NHORPAR,
              const std::vector<int>& ITYPHPAR, const std::vector<int>& IHORPAR,
              const std::vector<int>& LMAXHOR, const std::vector<int>& NCOEFHOR,
              const std::vector<std::vector<float>>& XLASPL, const std::vector<std::vector<float>>& XLOSPL,
              const std::vector<std::vector<float>>& XRASPL, const std::vector<std::vector<int>>& IXLSPL,
              const std::vector<std::vector<float>>& COEF, const std::vector<std::string>& HSPLFILE,
              const std::string& REFMODEL, const std::string& KERNSTRI, const std::vector<std::string>& DESCKERN);

} // namespace RD3DMODL__genmod

#endif // RD3DMODL_GENMOD_H

File: data/code_pairs/fortran/1372.f90
MODULE RD3DMODL__genmod
  IMPLICIT NONE
CONTAINS
  SUBROUTINE RD3DMODL(FILENAME, IERROR, NMODKERN, NHORPAR, ITYPHPAR, IHORPAR, LMAXHOR, NCOEFHOR, XLASPL, XLOSPL, XRASPL, IXLSPL, COEF, HSPLFILE, REFMODEL, KERNSTRI, DESCKERN)
    CHARACTER(LEN=*), INTENT(IN) :: FILENAME
    INTEGER, INTENT(INOUT) :: IERROR
    INTEGER, INTENT(IN) :: NMODKERN, NHORPAR
    INTEGER, DIMENSION(2), INTENT(IN) :: ITYPHPAR
    INTEGER, DIMENSION(200), INTENT(IN) :: IHORPAR
    INTEGER, DIMENSION(2), INTENT(IN) :: LMAXHOR
    INTEGER, DIMENSION(2), INTENT(IN) :: NCOEFHOR
    REAL, DIMENSION(2000,2), INTENT(IN) :: XLASPL, XLOSPL, XRASPL
    INTEGER, DIMENSION(2000,2), INTENT(IN) :: IXLSPL
    REAL, DIMENSION(2000,200), INTENT(IN) :: COEF
    CHARACTER(LEN=80), DIMENSION(2), INTENT(IN) :: HSPLFILE
    CHARACTER(LEN=80), INTENT(IN) :: REFMODEL, KERNSTRI
    CHARACTER(LEN=40), DIMENSION(200), INTENT(IN) :: DESCKERN
    
    PRINT *, "RD3DMODL subroutine called"
    IERROR = 0
  END SUBROUTINE RD3DMODL

END MODULE RD3DMODL__genmod